util/http/http.h

```diff
@@ -26,6 +26,7 @@ typedef struct http_method_s {
 } http_method_t;
 
 typedef struct http_server_options_s {
+  char addr[16];
```

Comment:
Tested with --http-addr=stringthatisreallytoolong, the program will segfault. With --http-addr=127.0.0.1:8090, it doesn't crash, but probably doesn't do what the user would expect using that (maybe the sscanf that parses the option overrides "port" and "maxcons" with garbage values ?)

That's not really a bug with this change, but I think util/opts/opts.c should consider option->size when parsing options with sscanf.

These memory safety problems are a good reason why not listening on INADDR_ANY seems like a great idea !

Reply:
Yeah. I should not reinvent a wheel, this project started as pure ANSI-C, but then it had to evolve to support C++. Now, it seems that it should be rust 📦

Reply:
Yeah, I was looking for a way to limit it for authentication reasons, but when I saw the memory issues I was even more glad I decided to limit it! Would be awesome to see this in Rust.